libs/algo/odc/algo/_masking.py

```diff
@@ -458,17 +579,87 @@ def _first_valid_np(*aa: np.ndarray, nodata: Union[float, int, None] = None) ->
     return out
 
 
-def _choose_with_custom_op(x: xr.DataArray, op) -> xr.DataArray:
+def _fuse_min_np(*aa: np.ndarray) -> np.ndarray:
+    """
+    Element wise min (propagates NaN values)
+    """
+    out = aa[0].copy()
+    for a in aa[1:]:
+        out = np.minimum(out, a, out=out)
+    return out
+
+
+def _fuse_max_np(*aa: np.ndarray) -> np.ndarray:
+    """
+    Element wise max (propagates NaN values)
+    """
+    out = aa[0].copy()
+    for a in aa[1:]:
+        out = np.maximum(out, a, out=out)
+    return out
+
+
+def _fuse_and_np(*aa: np.ndarray) -> np.ndarray:
+    """
+    Element wise bit and
+    """
+    assert len(aa) > 0
+    out = aa[0].copy()
+    for a in aa[1:]:
+        out &= a
```

Comment:
Not an issue, but can rely on numpy.bitwise_and.

Reply:
I'm not sure if the equivalent function based implementation is much clearer than an operator based one

```python
    out = aa[0].copy()
    for a in aa[1:]:
        np.bitwise_and(out, a, out=out)
```